package.json

```diff
@@ -56,13 +56,14 @@
     "yup": "^0.27.0"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "NODE_ENV=development npm run update-runtime-env && react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "NODE_ENV=test npm run update-runtime-env && react-scripts test",
```

Comment:
I'll try to humanize the changes that `NODE_ENV` goes through when this script is called.

1. `process.env.NODE_ENV` is set to `test`
2. `update-runtime-env` is invoked which saves the current `NODE_ENV` into the `env-config.js` file
3. CRA's `test` script is invoked, which sets `process.env.NODE_ENV` to `test` again
4. During the test command the `setupTests.ts` file is executed which adds the environment from `env-config` into the window object

I'm not sure, but I could imagine that this would be hard to reason for someone who is familiar with CRA, but not with the repo. I wonder if we should support such devs' understanding with comments (etc)?

Reply:
`react-scripts test` implicitly sets NODE_ENV to test. Issue is, that at that time there already needs to be the valid configuration for test, and to create that env needs to be given.

NODE_ENV is used in the code, so it has to go to the file (at serviceWorker and authService)